cloudinit/distros/__init__.py

```diff
@@ -77,12 +78,26 @@ class Distro(metaclass=abc.ABCMeta):
     # subclasses
     shutdown_options_map = {'halt': '-H', 'poweroff': '-P', 'reboot': '-r'}
 
+    _ci_pkl_version = 1
+
     def __init__(self, name, cfg, paths):
         self._paths = paths
         self._cfg = cfg
         self.name = name
         self.networking = self.networking_cls()
 
+    def _unpickle(self, ci_pkl_version: int) -> None:
+        """Perform deserialization fixes for Distro."""
+        if "networking" not in self.__dict__ or not self.networking.__dict__:
```

Comment:
There exist "version 0" `Distro` pickles which should not have this remediation applied to them, so we can't use the versioning for this check.